dipy/segment/fss.py

```diff
@@ -0,0 +1,291 @@
+            raise ValueError("radius should be smaller or equal to the given"
+                             "\n 'max_radius' in FastStreamlineSearch init")
+
+        # Resample query streamlines
+        q_slines = self._resample(streamlines)
+        q_nb_slines = len(q_slines)
+
+        # Compute streamlines barycenter
+        q_baryc = self._slines_barycenters(q_slines)
+
+        # Verify if each barycenter are inside the min max box
+        u_bin, binned_slines_ids = self._barycenters_binning(q_baryc)
+
+        # Adapting radius for L1 query: sqrt(3) = 1.73205080756887729..
+        # Rounded up for float32 precision to avoid error / false negative
+        l1_sum_dist = 1.73205081 * radius * self.nb_mpts
+
+        # Search for all similar streamlines
+        list_id = []
+        list_id_ref = []
+        list_dist = []
+        for i, bin_id in enumerate(u_bin):
+            if bin_id in self.bin_dict:
+                slines_id_ref, ref_tree = self.bin_dict[bin_id]
+                slines_id = binned_slines_ids[i]
+
+                mpts = self._slines_mean_points(q_slines[slines_id])
+
+                # Compute Tree L1 Query with mean-points
+                res = ref_tree.query_ball_point(mpts, l1_sum_dist, p=1)
+
+                # Refine distance with the complete
+                for s, ref_ids in enumerate(res):
+                    if ref_ids:
+                        cur_sline_id = slines_id[s]
+                        ref_sline_ids = slines_id_ref[ref_ids]
+                        d = mean_l2_func(q_slines[cur_sline_id],
+                                         self.ref_slines[ref_sline_ids])
+
+                        # Return all pairs within the radius
+                        in_dist_max = d < radius
+                        id_ref = ref_sline_ids[in_dist_max]
+                        id_s = np.full_like(id_ref, cur_sline_id)
+
+                        list_id.append(id_s)
+                        list_id_ref.append(id_ref)
+                        list_dist.append(d[in_dist_max])
+
+        # Combine all results in a coo sparse matrix
+        if len(list_id) > 0:
+            ids_in = np.hstack(list_id)
+            ids_ref = np.hstack(list_id_ref)
+            dist = np.hstack(list_dist)
+
+            if self.bidirectional:
+                flipped = ids_ref >= self.ref_nb_slines
+                ids_ref[flipped] -= self.ref_nb_slines
+                if use_negative:
+                    dist[flipped] *= -1.0
+
+            return coo_matrix((dist, (ids_in, ids_ref)),
+                              shape=(q_nb_slines, self.ref_nb_slines))
+
+        # No results, return an empty sparse matrix
+        return coo_matrix((q_nb_slines, self.ref_nb_slines))
+
+    def _resample(self, streamlines):
+        """Resample streamlines"""
+        s = np.zeros([len(streamlines), self.resampling, 3], dtype=np.float32)
+        for i, sline in enumerate(streamlines):
+            if len(sline) < 2:
```

Comment:
I tested it, because it fixes the error when havin a signle points (in the example it happens),
it does not link to "zero", it duplicates the points (numpy broadcast)
so it becomes a list of the same points, and my algorithms compares it like the rest (average euclidean distance)